packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -330,12 +326,13 @@ class AndroidLicenseValidator extends DoctorValidator {
   }
 
   Future<bool> _checkJavaVersionNoOutput() async {
-    final String? javaBinary = AndroidSdk.findJavaBinary(
+    final String? javaBinary = Java.find(
+      logger: _logger,
       androidStudio: _androidStudio,
       fileSystem: _fileSystem,
-      operatingSystemUtils: _operatingSystemUtils,
       platform: _platform,
-    );
+      processManager: _processManager,
+    )?.binaryPath;
     if (javaBinary == null) {
       return false;
     }
```

Comment:
For the canRun check we do on the next line, should we not do that during `Java.find()` and if you can't run it, keep falling back?

Reply:
Hmm, or should we tool exit because maybe the user should fix their JDK setup? I'm not sure, legitimately asking.

Reply:
> For the canRun check we do on the next line, should we not do that during `Java.find()` and if you can't run it, keep falling back?

Not sure what you mean here, though I probably should further migrate this code by using the `canRun` and `version` methods/properties on `Java`.

> Hmm, or should we tool exit because maybe the user should fix their JDK setup? I'm not sure, legitimately asking.

Hmm...I haven't looked at this code critically before. If we can't find the java version (or a well-formed Android SDK), we returned a `ValidationResult` with a `ValidationType` of `missing`, so I wonder what the `doctor` output would look like here. I'm just thinking out loud.

I can play around with this once I get home, where I have some a VM set up to test this kind of stuff.

Reply:
To unblock this PR you can add a TODO for this, if you like.